dollarbot.py

```diff
@@ -1,34 +1,63 @@
 from bs4 import BeautifulSoup
-import urllib3
-import re
 import requests
 import json
-import threading
-
-def dollar_poller():
-    dolar_venta = 0
-    threading.Timer(900.0, dollar_poller).start()
-    http = urllib3.PoolManager()
-    r = http.request('GET', 'http://dolarhoy.bullmarketbrokers.com/')
-    soup = BeautifulSoup(r.data, 'html.parser')
-    soupc = soup.find("h4")
-    soupv = soupc.findNext("h4")
-    dolar_c = re.search(r'\d\d\.\d\d', str(soupc)).group(0)
-    dolar_v = re.search(r'\d\d\.\d\d', str(soupv)).group(0)
-    dif = dolar_venta - float(dolar_v)
-
-    if float(dolar_v) > dolar_venta:
-        post_dollar_in_slack(dolar_c, dolar_v)
-        dolar_venta = dolar_v
-    elif dif > 0.10 or dif < -0.10:
-        post_dollar_in_slack(dolar_c, dolar_v)
-        dolar_venta = dolar_v
-
-def post_dollar_in_slack(compra, venta):
-    jason = {}
-    jason["text"] = "Compra: "+str(compra)+"  Venta: "+str(venta)
-    url = "https://hooks.slack.com/services/"
-    r = requests.post(url, data=json.dumps(jason))
-
-
-dollar_poller()
+import time
+
+config_file = "conf.yml"
+data_file = "data.txt"
+
+# gets prices from DolayHoy and reads data from files, then uses decision logic based on % threshold to notify
+def poll_prices():
+    r = requests.get('http://dolarhoy.bullmarketbrokers.com/')
+    soup = BeautifulSoup(r.content, 'html.parser')
+    soup_b = soup.find("h4")
+    soup_s = soup_b.findNext("h4")
+    buy = get_num(soup_b.getText())
+    sell = get_num(soup_s.getText())
+
+    config = read_conf()
+    slack_hook = config[0]
+    threshold = float(config[1])
+    last_price = float(read_data())
+    difference = sell - last_price
+    percentage = (difference / last_price) * 100.
+    if percentage > threshold:
+        notification = "Buy: ${:04.2f} Sell: ${:04.2f} - Percentage: {:04.1f}% Up".format(buy, sell, percentage)
+        post_to_slack(notification, slack_hook)
+        save_last_price(sell)
+    elif percentage < -threshold:
+        notification = "Buy: ${:04.2f} Sell: ${:04.2f} - Percentage: {:04.1f}% Down".format(buy, sell, percentage)
+        post_to_slack(notification, slack_hook)
+        save_last_price(sell)
+
+# returns a float from a given string -- works if there's only 1 dot in the string
+def get_num(s):
+    return float(''.join(char for char in s if char.isdigit() or char == '.'))
+
+# saves last_price on the file's first line
+def save_last_price(sell):
+    with open(data_file, "w") as f:
+        f.write(str(sell))
+
+# reads yaml configuration file and returns list with values
+def read_conf():
+    with open(config_file, "r") as f:
+        data = f.readlines()
+        # first line of the list would be Slack hook, second line would be percentage threshold
+    return data
+
+# reads data file with last saved sell price
+def read_data():
+    with open(data_file, "r") as f:
+        data = f.readline()
+    return data
+
+# posts a text to Slack
+def post_to_slack(text, webhook):
```

Comment:
consider adding a `channel` option here (maybe with a default like `#general`)

Reply:
Good idea. It could be added to the configuration file.